pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -454,9 +455,28 @@ class Utils {
     @Restricted(NoExternalUse.class)
     static <T> T instantiateDescribable(Class<T> c, Map<String, ?> args) {
         DescribableModel<T> model = new DescribableModel<>(c)
+        // Special case for JENKINS-63499.
+        if (model != null && model.type == PasswordParameterDefinition.class && model.getParameter("defaultValueAsSecret") != null) {
+            args = transformPasswordParameterDefinitionArgs(args)
+        }
         return model?.instantiate(args)
     }
 
+    /**
+     * Convert structure of password parameter definition arguments to fix JENKINS-63499.
+     */
+    private static <T> Map<String, Object> transformPasswordParameterDefinitionArgs(Map<String, ?> args) {
+        Map<String, Object> newArgs = new HashMap<>()
+        for (Map.Entry<String, ?> entry : args.entrySet()) {
+            if (entry.getKey().equals("defaultValue") && entry.getValue() instanceof String) {
+                newArgs.put("defaultValueAsSecret", Secret.fromString((String)entry.getValue()))
```

Comment:
Copy all arguments, but replace `defaultValue: 'foo'` with `defaultValueAsSecret: Secret.fromString('foo')` so that data binding succeeds.